### PR TITLE
Add "show entry" action to download notifications 

### LINKED ISSUE #1129: Add "Go to series" on download queue items
### Describe your suggested feature

In the download queue (accessible by long or double pressing "updates" tab), each items has several menu options. With one of them being "cancel all for series" I think it would be sensible to add a button to go to the series in question. 

### Other details

I can work on this one, probably

### Acknowledgements

- [X] I will fill out all of the requested information in this form.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.16.5](https://github.com/mihonapp/mihon/releases/latest)**.

### PR DESCRIPTION
**Pull request by mm12**
Adds a "Show entry" action to the following download notifications, allowing user to restart the download or view the entry:
- Normal: When showing download progress 
- Errors:
  - insufficient space
  - generic chapter error
  - generic image error
- Warnings: None

Provides a meaningful way to address the root cause, and resolve #1129

### CODE REVIEW

**Comment by AntsyLich** [comment]
add it to dropdown or don't

**Comment by AntsyLich** [comment]
I don't want to see another icon on the ui

File: app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadNotifier.kt [hunk 1]
```diff
@@ -83,6 +83,12 @@ internal class DownloadNotifier(private val context: Context) {
                     context.stringResource(MR.strings.action_pause),
                     NotificationReceiver.pauseDownloadsPendingBroadcast(context),
                 )
+                addAction(
+                    R.drawable.ic_book_24dp,
+                    context.stringResource(MR.strings.action_show_manga),
+
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadNotifier.kt:89]:
```suggestion
```

File: app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadNotifier.kt [hunk 2]
```diff
@@ -160,16 +166,25 @@ internal class DownloadNotifier(private val context: Context) {
...
             setStyle(NotificationCompat.BigTextStyle().bigText(reason))
             setSmallIcon(R.drawable.ic_warning_white_24dp)
             setAutoCancel(true)
             clearActions()
             setContentIntent(NotificationHandler.openDownloadManagerPendingActivity(context))
+            if (mangaId != null) {
+                addAction(
+                    R.drawable.ic_book_24dp,
+                    context.stringResource(MR.strings.action_show_manga),
+
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadNotifier.kt:184]:
```suggestion
```

File: app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadNotifier.kt [hunk 3]
```diff
@@ -198,6 +214,14 @@ internal class DownloadNotifier(private val context: Context) {
             setSmallIcon(R.drawable.ic_warning_white_24dp)
             clearActions()
             setContentIntent(NotificationHandler.openDownloadManagerPendingActivity(context))
+            if (mangaId != null) {
+                addAction(
+                    R.drawable.ic_book_24dp,
+                    context.stringResource(MR.strings.action_show_manga),
+
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadNotifier.kt:221]:
```suggestion
```

File: app/src/main/java/eu/kanade/tachiyomi/data/notification/NotificationReceiver.kt [hunk 4]
```diff
@@ -484,6 +486,26 @@ class NotificationReceiver : BroadcastReceiver() {
...
+         * @param context context of application
+         * @param mangaId id of the entry to open
+         */
+        internal fun openEntryPendingActivity(context: Context, mangaId: Long): PendingIntent {
+            val newIntent =
+                Intent(context, MainActivity::class.java).setAction(Constants.SHORTCUT_MANGA)
+                    .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+                    .putExtra(Constants.MANGA_EXTRA, mangaId)
+                    .putExtra("notificationId", mangaId.hashCode())
+            return PendingIntent.getActivity(
```
Comment (AntsyLich) [inline app/src/main/java/eu/kanade/tachiyomi/data/notification/NotificationReceiver.kt:501]:
```suggestion
            val newIntent = Intent(context, MainActivity::class.java).setAction(Constants.SHORTCUT_MANGA)
                .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
                .putExtra(Constants.MANGA_EXTRA, mangaId)
                .putExtra("notificationId", mangaId.hashCode())
                    
            return PendingIntent.getActivity(
```

**Review by AntsyLich — CHANGES_REQUESTED** [review]

File: app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadNotifier.kt (hunk 3 above)
Comment (MajorTanya) [reply app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadNotifier.kt:221]: These suggestions are only removing blank lines

Comment (MajorTanya) [reply app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadNotifier.kt:221]: > Ah ok. It is very hard to tell in both the "review files" and the "conversations" interface.

grafik
this is the box for the actual change, the code bit above it is the context for the suggestion

**Pull request merged**